Core/Src/main.c

```diff
@@ -87,9 +102,15 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
 
-  /* USER CODE END 2 */
+  /* initialize PWM timer */
+  HAL_TIM_PWM_Init(&htim1);
```

Comment:
you may have initialized the peripheral but you didn’t start it, so at this point, no signal is being generated… I’m fairly sure that you don’t need to initialize the timer bec it is already initialized on line 106… starting the PWM timer here should be sufficient